Microsoft.HealthVault.Fhir/Codings/FhirCodesToHealthVault.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Get Real Health.  All rights reserved.
+// MIT License
+// Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Hl7.Fhir.Model;
+using Microsoft.HealthVault.Fhir.Constants;
+using Microsoft.HealthVault.ItemTypes;
+
+namespace Microsoft.HealthVault.Fhir.Codings
+{
+    /// <summary>
+    /// Fhir have CodeableConcept
+    /// with a list of codings and/or text representation
+    /// Coding have a system, version, code and display
+    /// HV have CodableValue 
+    /// with a list of codedValue and Text
+    /// CodedValue has value, vocabularyName, family and version
+    /// HV also have a concept of vocabularyItem
+    /// with vocabularyName, version, family, value
+    /// displaytext and abbrevationtext
+    /// </summary>
+    public static class FhirCodesToHealthVault
+    {
+        public static CodableValue GetCodableValue(this CodeableConcept codeableConcept)
```

Comment:
There's now an extension method for "ToCodableValue" that should be able to handle most cases for doing this. If there's a case that's not been accounted for, please update that extension method to take care of it.